src/NuGet.Core/NuGet.CommandLine.XPlat/ReportRenderers/Models/ListPackageProjectModel.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Evaluation;
+using NuGet.CommandLine.XPlat.Utility;
+using NuGet.ProjectModel;
+
+namespace NuGet.CommandLine.XPlat
+{
+    /// <summary>
+    /// Calculated project data model for list report
+    /// </summary>
+    internal class ListPackageProjectModel
+    {
+        private const string ProjectNameMSbuildProperty = "MSBuildProjectName";
+        internal List<ReportProblem> ProjectProblems { get; } = new List<ReportProblem>();
+        internal string ProjectPath { get; private set; }
+        // Original packages
+        internal IEnumerable<FrameworkPackages> Packages { get; private set; }
+        // Calculated project model data for each targetframeworks
+        internal List<ListPackageReportFrameworkPackage> TargetFrameworkPackages { get; private set; }
+        internal string HttpSourceWarning { get; private set; }
+        internal string ProjectName { get; private set; }
+        internal Project Project { get; }
+        internal ListPackageReportModel ReportModel { get; }
+        internal bool AutoReferenceFound { get; private set; }
+
+        public ListPackageProjectModel(string projectPath, ListPackageReportModel reportModel, Project project)
+        {
+            ProjectPath = projectPath;
+            ReportModel = reportModel;
+            Project = project;
+            ProjectName = Project.GetPropertyValue(ProjectNameMSbuildProperty);
```

Comment:
Passing around Project is imo a leaky abstraction. 

Calculate the project name on the outside and initialize this object with it.

Reply:
+1. In (my) "ideal" MVC pattern, models are POCO with no logic, logic goes in the controller, and views just format the output appropriately.  Often there are practical constraints that make such an ideal difficult, but I don't think this is one of those cases.

Reply:
Sure, removed the `Project` and any other logics from `ListPackageProjectModel.cs` file.